SYCL/ESIMD/api/functional/type_coverage.hpp

```diff
@@ -86,28 +91,42 @@ template <tested_types required> auto get_tested_types() {
   }
 }
 
+auto get_all_vector_lengths() { return values_pack<1, 8, 16, 32>(); }
+
 // Run action for each of types given by named_type_pack instance
-template <template <typename, typename...> class action,
-          typename... actionArgsT, typename... types, typename... argsT>
-inline bool for_all_types(const named_type_pack<types...> &typeList,
-                          argsT &&... args) {
+template <template <typename, int, typename...> class Action, int N,
+          typename... ActionArgsT, typename... TypesT, typename... ArgsT>
+inline bool for_all_types(const named_type_pack<TypesT...> &type_list,
+                          ArgsT &&... args) {
   bool passed{true};
 
   // run action for each type from types... parameter pack
-  size_t typeNameIndex = 0;
-
-  int packExpansion[] = {
-      (passed &= action<types, actionArgsT...>{}(std::forward<argsT>(args)...,
-                                                 typeList.names[typeNameIndex]),
-       ++typeNameIndex,
-       0 // Dummy initialization value
-       )...};
-  // Every initializer clause is sequenced before any initializer clause that
-  // follows it in the braced-init-list. Every expression in comma operator is
-  // also strictly sequenced. So we can use increment safely. We still should
-  // discard dummy results, but this initialization should not be optimized out
-  // due side-effects
-  static_cast<void>(packExpansion);
+  size_t type_name_index = 0;
+
+  ((passed &= Action<TypesT, N, ActionArgsT...>{}(
+        std::forward<ArgsT>(args)..., type_list.names[type_name_index]),
+    ++type_name_index),
+   ...);
+
+  return passed;
+}
+
+// Calls for_all_types for each vector length by values_pack instance
+template <template <typename, int, typename...> class Action,
+          typename... ActionArgsT, typename... TypesT, int... Ints,
+          typename... ArgsT>
+inline bool for_types_and_dims(const named_type_pack<TypesT...> &type_list,
```

Comment:
```suggestion
inline bool for_all_types_and_dims(const named_type_pack<TypesT...> &type_list,
```
Once again, out of uniformity and to make the full set of combinations used explicit.

Reply:
I've reverted this comment [here](https://github.com/intel/llvm-test-suite/pull/588/commits/b359d36a0cf7e324f477dea39bd6f3736b648c55#diff-3bbc1ab1445c827282cc08d73f76b58cf553d9ced2c799ba46e6148583896d22R133)